stock_picking_import_serial_number/wizard/import_serial_number_wizard.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2022 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import base64
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+import xlrd
+
+
+class StockPickingImportSerialNumber(models.TransientModel):
+    _name = "stock.picking.import.serial.number.wiz"
+    _description = "Import S/N wizard"
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        if (
+            self.env.context.get("active_ids")
+            and self.env.context.get("active_model") == "stock.picking"
+        ):
+            pickings = self.env["stock.picking"].browse(
+                self.env.context.get("active_ids")
+            )
+            if pickings.exists():
+                res.update({"picking_ids": [(6, 0, pickings.ids)]})
+        return res
+
+    search_product_by_field = fields.Selection(
+        [("default_code", "Reference"), ("barcode", "Barcode")],
+        string="Search product by field",
+        default="default_code",
+    )
+    product_column_index = fields.Integer(string="Column index for product", default=0)
+    serial_column_index = fields.Integer(string="Column index for S/N", default="1")
+    picking_ids = fields.Many2many("stock.picking")
+    data_file = fields.Binary(string="File to import")
+    filename = fields.Char(string="Filename")
+    overwrite_serial = fields.Boolean()
+
+    def action_import(self):
+        if self.picking_ids.filtered(lambda p: p.picking_type_id.code != "incoming"):
+            raise UserError(_("You only can import S/N for incoming moves"))
+        if self.picking_ids.filtered(lambda p: not p.picking_type_id.use_create_lots):
+            raise UserError(
+                _(
+                    "You only can import S/N for picking operations with"
+                    " creation lots checked"
+                )
+            )
+        if not self.data_file:
+            raise UserError(_("You must upload file to import records"))
+        xl_workbook = False
+        xl_sheet = False
+        if self.filename.split(".")[1] in ["xls", "xlsx"]:
+            xl_workbook = xlrd.open_workbook(
+                file_contents=base64.b64decode(self.data_file)
+            )
+            xl_sheet = xl_workbook.sheet_by_index(0)
+            move_lines = self.picking_ids.mapped("move_line_ids").filtered(
```

Comment:
Shouldn't we be relaying in `move_lines` instead?

Reply:
I am working over stock.move.lines